### ISSUE TITLE
Support microsoft-ui-xaml stable version in unpackaged application

### ISSUE THREAD

**Issue by DeepthiCG** [microsoft/microsoft-ui-xaml#5180]
Hosting controls from microsoft-ui-xaml prerelease versions to **unpackaged** desktop application using XAML island is possible now.
But not possible to host the controls from microsoft-ui-xaml 2.5.0 release version. 
Is there any work around for supporting microsoft-ui-xaml 2.5.0 release version to **unpackaged** desktop applications, for production?

**Comment by StephenLPeters** [comment]
The release versions are framework packages, which we need for servicing. So I think the answer is no, although you could ship using a prerelease package.  @jevansaks to confirm?

**Comment by DeepthiCG** [comment]
Thank you for the reply. 
Could you confirm below doubts also?
Whether creating Sparse Package for the desktop app, solves this issue?
(https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/grant-identity-to-nonpackaged-apps)
Also whether there any impact for the desktop application which is currently working as unpackaged in market as part of creating a sparse package for it?

**Comment by jevansaks** [comment]
@drustheaxe to comment on the state of taking framework package dependencies from unpackaged apps. I don't know if sparse packages help -- I think they would help you register the dependent framework but it won't add the dependency to your unpackaged app's package graph at runtime. For that you need the Reunion APIs that @drustheaxe is working on.

**Comment by DeepthiCG** [comment]
Thank you for the reply.

Please confirm whether Reunion APIs are supported in WinUI2x stable versions? Or can be used only with WinUI3?

**Comment by DeepthiCG** [comment]
My understanding is that, as part of "MSIX Dynamic Dependencies" of Project Reunion/Windows App SDK , it is possible to access any frame work packaged content ( eg:WinUI2.5 stable) by an unpackaged application at runtime?. Hope this feature will be released as part of Windows App SDK 1.0  for production?

**Comment by DrusTheAxe** [comment]
> as part of "MSIX Dynamic Dependencies" of Project Reunion/Windows App SDK , it is possible to access any frame work packaged content ( eg:WinUI2.5 stable) by an unpackaged application at runtime?

Yes, but.

Can you? Sure! After calling `MddBootstrapInitialize()` to get access to the framework package you can call `MddTryCreatePackageDependency()` + `MddAddPackageDependency()` to get access to WinUI2.5 or other framework packages.

But...Windows won't know your app is using the WinUI2.5 framework package so nothing stops the deployment stack from servicing the package (update, remove, repair, reset, ...). The Dynamic Dependency Lifetime Manager (DDLM) package is how the Dynamic Dependency API tells Windows the Project Reunion framework package is in use. You'd need an equivalent package for WinUI2.5 to not run afoul of servicing.

The DDLM package is conceptually simple. Windows needs to see a packaged process where its package's `appxmanifest.xml` has a `<PackageDependency Name="WinUI2.5"...>. When servicing WinUI2.5 the Deployment stack sees if there's running processes using the framework and if so, don't service the framework in ways that impact running processes (typically means add e.g. v2.5.0.1 but don't remove v2.5.0.0). The DDLM accomplish this by being a Main package containing a Packaged COM OOP Server, and when `MddBootstrapInitialize()` is called it does a `CoCreateInstance()` on that COM object and hangs onto it until `MddBootstrapShutdown()`. Thus the COM server is a process with the package's identity and running while your process is using the Project Reunion framework package.

The DDLM package is specific to Project Reunion. You'd need an equivalent but different package for WinUI2.5 (or other framework packages).

Long winded way of saying it's possible. One option is to fork the DDLM support to be a 'starter kit' to create <insert-framework-name-here>DDLMs. Another is to create an equivalent package using the same techniques. Regardless it's possible, but a healthy amount of work.

Is this something you'd like to see?

What frameworks interest you (besides Project Reunion :P) for use by unpackaged apps? Is it just WinUI2.5? Other versions? Other frameworks?

>Hope this feature will be released as part of Windows App SDK 1.0 for production?

That's the plan :-) The [roadmap](Bug 34287176: [RS3/RS2 Upgrade][Compat][Windows Hello PIN][Surface Book 2]:-CredentialsUIBroken.exe/Windows.UI.XamlHost.dll crashes while trying to set windows Hello PIN) doesn't make it clear but where  you see "Unpackaged desktop apps" that means Dynamic Dependencies is in play (the defining tech making unpackaged apps support possible :-)

There's docs about Dynamic Dependencies but they're not directly linked or mentioned in the roadmap :-( I've pinged some folks about making that more apparent and discoverable.

**Comment by DeepthiCG** [comment]
Thank you for the details.
Our plan is to use WinUI2x stable version in a legacy MFC application using XAML island. Since the application is unpackaged, currently it is not possible and only prerelease versions can be used. 
From above explanation, understood like DDLM package is specific to Project Reunion and not support WinUI2x packages. Since WinUI3 is a part of project reunion, hope it can be used in unpackaged application using DDLM package after XAML Island support in WinUI3?

**Comment by DrusTheAxe** [comment]
> From above explanation, understood like DDLM package is specific to Project Reunion and not support WinUI2x packages. Since WinUI3 is a part of project reunion, hope it can be used in unpackaged application using DDLM package after XAML Island support in WinUI3?

Are you asking if WinUI3 will support unpackaged apps? If a packaged WinUI2 app can use a future WinUI3 XAML Island feature? Or...?

**Comment by DeepthiCG** [comment]
Yes, my question is , whether WinUI3 stable versions can be used in MFC unpackaged application through XAML Island?
